src/actions/publish.js

```diff
@@ -0,0 +1,144 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { compress } from '../utils/gz';
+import { uploadBinaryProfileData } from '../profile-logic/profile-store';
+import { serializeProfile } from '../profile-logic/process-profile';
+import { sendAnalytics } from '../utils/analytics';
+import { getProfile } from '../selectors/profile';
+import { getUrlState } from '../selectors/url-state';
+import {
+  getAbortFunction,
+  getUploadPhase,
+  getUploadGeneration,
+} from '../selectors/publish';
+import { urlFromState } from '../app-logic/url-handling';
+import { profilePublished } from './app';
+import urlStateReducer from '../reducers/url-state';
+
+import type { Action, ThunkAction } from '../types/store';
+import type { UploadState } from '../types/state';
+import type { CheckedSharingOptions } from '../types/actions';
+
+export const toggleCheckedSharingOptions = (
```

Comment:
I'm not happy with code coverage here yet, I filed a follow-up. I figure this code will change with follow-ups, so I didn't want to over-invest in unit tests. https://github.com/firefox-devtools/profiler/issues/1895